Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2994,6 +3000,40 @@ extension BrowserViewController {
             NavigationPath.handle(nav: path, with: self)
         }
     }
+    
+    func requestReviewIfNecessary() {
```

Comment:
Can we move this logic outside of bvc?
I think `BraveShared/Analytics/AppReview.swift` would be the best.

And we could add some unit tests to it, add `date: Date = Date()` argument to the method to make testing easier.
See `DAU.swift` and `DAUTests.swift` it should be kind of similar logic.

The only thing we should do in bvc is to request the review so perhaps this method could also return a boolean